base/deprecated.jl

```diff
@@ -2097,6 +2097,20 @@ end
     @deprecate chol!(x::Number, uplo) chol(x) false
 end
 
+
+# issue #16307
+@deprecate finalizer(o, f::Function) finalizer(f, o)
+# This misses other callables but they are very rare in the wild
+
+@deprecate finalizer(o::T, f::Ptr{Void}) where {T} finalizer(f, o)
```

Comment:
`T where T` is `Any`, which is the same as applying no type annotation to the argument, so this could be simplified to match the deprecation above it.